src/common/CircleGameObject.cpp

```diff
@@ -1,18 +1,35 @@
 #include "CircleGameObject.hpp"
 
 
-CircleGameObject::CircleGameObject(unsigned int id, sf::Vector2f position, float size, sf::Color color)
-    : GameObject(id), mCircle(std::shared_ptr<Circle>(new Circle(size, position, color))) {}
+CircleGameObject::CircleGameObject(unsigned int id, sf::Vector2f position, float radius,
+    sf::Color color, std::shared_ptr<PhysicsController> pc)
+    : GameObject(id),
+      mCircle(std::shared_ptr<Circle>(new Circle(radius, position, color))),
+      mCircleRigidBody(pc->createCRB(radius, position)) {}
 
 std::shared_ptr<Circle> CircleGameObject::getCircle() {
     return mCircle;
 }
 
-std::vector<std::shared_ptr<Circle>> CircleGameObject::getCircles(
-    std::vector<std::shared_ptr<CircleGameObject>>& circle_game_objects) {
-    std::vector<std::shared_ptr<Circle>> circles;
-    std::transform(
-        circle_game_objects.begin(), circle_game_objects.end(), std::back_inserter(circles),
-        [](std::shared_ptr<CircleGameObject> cgo){return cgo->getCircle();});
-    return circles;
-};
+void CircleGameObject::setActive(bool is_active) {
+    mIsActive = is_active;
+    mCircleRigidBody->setActive(is_active);
+}
+
+void CircleGameObject::setVelocity(sf::Vector2f velocity) {
+    mCircleRigidBody->setVelocity(velocity);
```

Comment:
so is the circle rigid body shared by both the physics controller and the circle game object? Seems like we might end up leaking memory if we're not careful

Reply:
I guess I'm wondering if the physics controller should be the only manager and when we need to update a certain circles properties, we call the physics controller with that circles ID. single owner seems less error prone and easier to reason about

Reply:
physics controller is the owner, that's where it's initialized with `new` and kept in a vector `mCircleRigidBodies`. 
CircleGameObject just has a pointer to that object. Ideally it wouldn't even be allowed to delete it, but it's not really its job to delete it.
Just in case I've made the pointer a `std::shared_ptr<CircleRigidBody>` which is a "smart" pointer, but to be honest I haven't looked into this enough to be certain. 
At some point I really need to look into valgrind and define reconstructors.

That being said I don't think this design is flawed.

Reply:
It would be verbose and non-sustainable to create methods in PhysicsController with which you modify `CircleRigidBody`s... that's why I have the pointer shared so `CircleGameObject` can call things like `setVelocity`.

Reply:
sounds good, I think it's fine for now, if we see errors we can reconsider